### DISCUSSION TITLE
pydantic.errors.PydanticImportError: `pydantic:Required` has been removed in V2.

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#10021] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated
from pydantic import Required

from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/items/")
async def read_items(q: Annotated[str | None, Query(max_length=50)]):
    results = {"items": [{"item_id": "Foo"}, {"item_id": "Bar"}]}
    if q:
        results.update({"q": q})
    return results
```

### Description

I just started learning fastapi and while going through the documentation. I encontered the following error

`pydantic.errors.PydanticImportError: `pydantic:Required` has been removed in V2.`

The error clearly says that "Required" has been removed from v2. The documentation needs to be updated

### Operating System

Windows

### FastAPI Version

0.101.0

### Pydantic Version

2.1.1

### Python Version

Python 3.11.4

**Comment by LiuJiewenTT** [comment]
I have same problem when building with pyinstaller 4.7, python 3.8.8. The built app also said it cannot initialize dll on starting. Confusing...
